presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PruneCrossJoinColumns.java

```diff
@@ -63,8 +63,6 @@ public Pattern getPattern()
                 .map(ImmutableSet::copyOf)
                 .map(dependencies ->
                         parent.replaceChildren(ImmutableList.of(
-                                joinNode.replaceChildren(ImmutableList.of(
-                                        restrictOutputs(idAllocator, joinNode.getLeft(), dependencies).orElse(joinNode.getLeft()),
-                                        restrictOutputs(idAllocator, joinNode.getRight(), dependencies).orElse(joinNode.getRight()))))));
+                                restrictBinaryChildrenOutputs(idAllocator, joinNode, dependencies, dependencies).get())));
```

Comment:
I am not sure if previous version wasn't better. Maybe I got used to restrictOutputs, and so it was easier to me understand what it does.

Anyway, please extract this unrelated change as separate commit.